components/admin_console/team_channel_settings/channel/channel_list.jsx

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import AbstractList from 'components/admin_console/team_channel_settings/abstract_list.jsx';
+import {browserHistory} from 'utils/browser_history';
+
+import ChannelRow from './channel_row';
+
+export default class ChannelList extends AbstractList {
```

Comment:
Too much code to change it now, but inheritance and components are not very good friends, I prefer other approaches like composition or HOC if needed, but... as I said, this is a bit late now, too much code here. I would prefer to rethink it later.